AK/SipHash.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+
+namespace AK {
+
+// Ported from the SipHash reference implementation, released to the public domain:
+// https://github.com/veorq/SipHash/blob/eee7d0d84dc7731df2359b243aa5e75d85f6eaef/siphash.c
+// The standard is SipHash-2-4, but we use 1-3 for a little more speed.
+// Cryptography should use 4-8 for conservative security.
+// Make sure to always use the same SipHash instance for a container so that the random key stays the same!
+template<size_t message_block_rounds = 1, size_t finalization_rounds = 3>
+struct SipHash {
+    unsigned operator()(u64 input);
+};
```

Comment:
This is really awkwardly factored, and I'm not sure why this implementation detail needs to be exposed to users at all?